File: iterations/tabletest/next/iteration-2/eval-28-convert-from-methodsource/outputs/src/test/java/com/example/ShippingCostCalculatorTest.java
package com.example;

import org.tabletest.junit.Description;
import org.tabletest.junit.TableTest;
import org.tabletest.junit.TypeConverter;

import java.math.BigDecimal;
import java.util.List;
import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class ShippingCostCalculatorTest {

    private final ShippingCostCalculator calculator = new ShippingCostCalculator();

    @Description("""
        Dimensions are fixed at [10, 10, 10] (well below the oversize threshold,
        and its dimensional weight of 0.2kg never exceeds the actual weights used
        here) so the rule under test is purely region + speed + actual weight.
        Dimensional weight override, the oversize fee, and package-option
        surcharges are covered in the dedicated tables below.
        """)
    @TableTest("""
        Scenario                | Region | Speed    | Weight | Base Rate?
        EU standard light       | EU     | standard | 0.5    | 5.00
        EU standard medium      | EU     | standard | 3.0    | 7.50
        EU standard heavy       | EU     | standard | 10.0   | 12.50
        EU standard very heavy  | EU     | standard | 25.0   | 20.00
        EU express light        | EU     | express  | 0.5    | 8.00
        EU express medium       | EU     | express  | 3.0    | 12.00
        US standard light       | US     | standard | 0.5    | 7.00
        US express heavy        | US     | express  | 10.0   | 30.00
        """)
    void calculatesBaseRateByRegionSpeedAndWeight(String region, String speed, double weight, BigDecimal baseRate) {
        ShippingZone zone = new ShippingZone(region, speed);
        BigDecimal result = calculator.calculateShippingCost(zone, weight, List.of(10, 10, 10), new PackageOptions(), Carrier.DHL);
        assertEquals(0, result.compareTo(baseRate));
    }

    @Description("""
        Effective billable weight is the greater of actual weight and dimensional
        weight (volume / 5000). Zone fixed to EU standard since the rate lookup
        itself is covered by calculatesBaseRateByRegionSpeedAndWeight.
        """)
    @TableTest("""
        Scenario                 | Weight | Dimensions   | Total Cost?
        Actual weight wins       | 20.0   | [10, 10, 10] | 20.00
        Dimensional weight wins  | 1.0    | [70, 50, 10] | 12.50
        """)
    void effectiveWeightIsGreaterOfActualOrDimensionalWeight(double weight, List<Integer> dimensions, BigDecimal totalCost) {
        ShippingZone zone = new ShippingZone("EU", "standard");
        BigDecimal result = calculator.calculateShippingCost(zone, weight, dimensions, new PackageOptions(), Carrier.DHL);
        assertEquals(0, result.compareTo(totalCost));
    }

    @Description("""
        A flat $10.00 oversize fee applies when any single dimension exceeds
        100 (100 itself is not oversize). Weight fixed at 3.0kg and zone fixed
        to EU standard (base rate 7.50) to isolate this surcharge.
        """)
    @TableTest("""
        Scenario                          | Dimensions   | Total Cost?
        Longest side at threshold         | [100, 5, 5]  | 7.50
        Longest side just over threshold  | [101, 5, 5]  | 17.50
        Large oversize package            | [120, 5, 5]  | 17.50
        """)
    void addsFlatFeeWhenAnyDimensionExceedsOversizeThreshold(List<Integer> dimensions, BigDecimal totalCost) {
        ShippingZone zone = new ShippingZone("EU", "standard");
        BigDecimal result = calculator.calculateShippingCost(zone, 3.0, dimensions, new PackageOptions(), Carrier.DHL);
        assertEquals(0, result.compareTo(totalCost));
    }

    @Description("""
        Weight fixed at 3.0kg, dimensions [30, 20, 15], zone EU standard (base
        rate 7.50). The hazmat fee is added before the fragile multiplier; the
        insurance premium is added after. The insurance minimum-premium floor
        is covered separately in insurancePremiumUsesGreaterOfCalculatedPremiumOrMinimum.
        Open question: this table does not exercise hazmat combined with
        fragile or insurance — ordering would follow the same rule but is
        untested here.
        """)
    @TableTest("""
        Scenario             | Options                             | Total Cost?
        No special options   | [:]                                  | 7.50
        Fragile package      | [fragile: true]                      | 8.625
        Insured package       | [insuredValue: 500]                  | 10.50
        Fragile and insured  | [fragile: true, insuredValue: 200]    | 11.625
        Hazmat handling      | [handling: hazmat]                    | 15.50
        """)
    void appliesPackageOptionSurcharges(PackageOptions options, BigDecimal totalCost) {
        ShippingZone zone = new ShippingZone("EU", "standard");
        BigDecimal result = calculator.calculateShippingCost(zone, 3.0, List.of(30, 20, 15), options, Carrier.DHL);
        assertEquals(0, result.compareTo(totalCost));
    }

    @Description("""
        Insurance premium is insuredValue x 0.006, floored at a $3.00 minimum.
        Weight fixed at 3.0kg, dimensions [30, 20, 15], zone EU standard (base
        rate 7.50).
        """)
    @TableTest("""
        Scenario                      | Insured Value | Total Cost?
        Premium below minimum floor   | 200           | 10.50
        Premium at minimum threshold  | 500           | 10.50
        Premium above minimum         | 600           | 11.10
        """)
    void insurancePremiumUsesGreaterOfCalculatedPremiumOrMinimum(BigDecimal insuredValue, BigDecimal totalCost) {
        ShippingZone zone = new ShippingZone("EU", "standard");
        PackageOptions options = new PackageOptions();
        options.setInsuredValue(insuredValue);
        BigDecimal result = calculator.calculateShippingCost(zone, 3.0, List.of(30, 20, 15), options, Carrier.DHL);
        assertEquals(0, result.compareTo(totalCost));
    }

    @Description("""
        Carrier does not influence price under the current model. Fixed
        inputs elsewhere in this class default to Carrier.DHL since it has
        no effect on the result.
        """)
    @TableTest("""
        Scenario                           | Carrier            | Total Cost?
        Price is the same for all carriers | {DHL, UPS, FEDEX}  | 12.00
        """)
    void totalCostIsIndependentOfCarrier(Carrier carrier, BigDecimal totalCost) {
        ShippingZone zone = new ShippingZone("EU", "express");
        BigDecimal result = calculator.calculateShippingCost(zone, 3.0, List.of(30, 20, 15), new PackageOptions(), carrier);
        assertEquals(0, result.compareTo(totalCost));
    }

    @TypeConverter
    public static PackageOptions parseOptions(Map<String, String> config) {
        PackageOptions options = new PackageOptions();
        if (config == null) {
            return options;
        }
        if (config.containsKey("fragile")) {
            options.setFragile(Boolean.parseBoolean(config.get("fragile")));
        }
        if (config.containsKey("insuredValue")) {
            options.setInsuredValue(new BigDecimal(config.get("insuredValue")));
        }
        if (config.containsKey("handling")) {
            options.setHandling(config.get("handling"));
        }
        return options;
    }
}

File: iterations/tabletest/next/iteration-2/eval-26-convert-from-kotest/outputs/src/test/kotlin/com/example/ShippingCostCalculatorTest.kt
package com.example

import org.junit.jupiter.api.Assertions.assertEquals
import org.tabletest.junit.Description
import org.tabletest.junit.TableTest
import java.math.BigDecimal

class ShippingCostCalculatorTest {

    private val calculator = ShippingCostCalculator()

    @TableTest("""
        Scenario                    | Region | Speed    | Weight | Dimensions   | Base Cost?
        EU standard, light          | EU     | standard | 0.5    | [20, 15, 10] | 5.00
        EU standard, medium         | EU     | standard | 3.0    | [30, 20, 15] | 7.50
        EU standard, heavy          | EU     | standard | 10.0   | [40, 30, 20] | 12.50
        EU standard, very heavy     | EU     | standard | 25.0   | [50, 40, 30] | 20.00
        EU express, light           | EU     | express  | 0.5    | [20, 15, 10] | 8.00
        EU express, medium          | EU     | express  | 3.0    | [30, 20, 15] | 12.00
        US standard, light          | US     | standard | 0.5    | [20, 15, 10] | 7.00
        US express, heavy           | US     | express  | 10.0   | [40, 30, 20] | 30.00
        Dimensional weight override | EU     | standard | 1.0    | [70, 50, 10] | 12.50
        """)
    fun calculatesBaseRateByZoneWeightAndDimensions(
        region: String, speed: String, weight: Double, dimensions: List<Int>, baseCost: BigDecimal
    ) {
        val cost = calculator.calculateShippingCost(
            ShippingZone(region, speed), weight, dimensions, PackageOptions(), Carrier.DHL
        )
        assertEquals(0, baseCost.compareTo(cost))
    }

    @Description("""
        All rows use EU standard shipping at 3.0 kg, giving a base cost of 7.50
        before the oversize surcharge. A package is oversize if any single
        dimension exceeds 100.
        """)
    @TableTest("""
        Scenario                      | Dimensions   | Total Cost?
        At oversize threshold         | [100, 5, 5]  | 7.50
        Just over oversize threshold  | [101, 5, 5]  | 17.50
        Well over oversize threshold  | [120, 5, 5]  | 17.50
        """)
    fun addsOversizeSurchargeForLargeDimensions(dimensions: List<Int>, totalCost: BigDecimal) {
        val cost = calculator.calculateShippingCost(
            ShippingZone("EU", "standard"), 3.0, dimensions, PackageOptions(), Carrier.DHL
        )
        assertEquals(0, totalCost.compareTo(cost))
    }

    @Description("""
        All rows use EU standard shipping at 3.0 kg with non-oversize dimensions,
        giving a base cost of 7.50 before the hazmat surcharge.
        """)
    @TableTest("""
        Scenario             | Handling | Total Cost?
        No special handling  |          | 7.50
        Hazmat handling      | hazmat   | 15.50
        """)
    fun addsHazmatHandlingSurcharge(handling: String?, totalCost: BigDecimal) {
        val options = PackageOptions(handling = handling)
        val cost = calculator.calculateShippingCost(
            ShippingZone("EU", "standard"), 3.0, listOf(30, 20, 15), options, Carrier.DHL
        )
        assertEquals(0, totalCost.compareTo(cost))
    }

    @Description("""
        All rows use EU standard shipping at 3.0 kg with non-oversize dimensions,
        giving a base cost of 7.50 before the fragile surcharge.
        """)
    @TableTest("""
        Scenario     | Fragile | Total Cost?
        Not fragile  | false   | 7.50
        Fragile      | true    | 8.625
        """)
    fun addsFragileSurcharge(fragile: Boolean, totalCost: BigDecimal) {
        val options = PackageOptions(isFragile = fragile)
        val cost = calculator.calculateShippingCost(
            ShippingZone("EU", "standard"), 3.0, listOf(30, 20, 15), options, Carrier.DHL
        )
        assertEquals(0, totalCost.compareTo(cost))
    }

    @Description("""
        All rows use EU standard shipping at 3.0 kg with non-oversize dimensions,
        giving a base cost of 7.50 before insurance. Insurance premium is the
        insured value times 0.6%, with a 3.00 minimum.
        """)
    @TableTest("""
        Scenario                            | Insured Value | Total Cost?
        No insurance                        |                | 7.50
        Insured value below minimum premium | 100            | 10.50
        Insured value at minimum premium    | 500            | 10.50
        Insured value above minimum premium | 1000           | 13.50
        """)
    fun addsInsuranceSurcharge(insuredValue: BigDecimal?, totalCost: BigDecimal) {
        val options = PackageOptions(insuredValue = insuredValue)
        val cost = calculator.calculateShippingCost(
            ShippingZone("EU", "standard"), 3.0, listOf(30, 20, 15), options, Carrier.DHL
        )
        assertEquals(0, totalCost.compareTo(cost))
    }

    @Description("""
        All rows use EU standard shipping at 3.0 kg with non-oversize dimensions
        (base cost 7.50) and are fragile, confirming the fragile multiplier is
        applied to the base cost only, before the insurance premium is added.
        """)
    @TableTest("""
        Scenario                                       | Insured Value | Total Cost?
        Fragile with insurance at minimum premium      | 200            | 11.625
        Fragile with insurance above minimum premium   | 1000           | 14.625
        """)
    fun combinesFragileAndInsuranceSurcharges(insuredValue: BigDecimal, totalCost: BigDecimal) {
        val options = PackageOptions(isFragile = true, insuredValue = insuredValue)
        val cost = calculator.calculateShippingCost(
            ShippingZone("EU", "standard"), 3.0, listOf(30, 20, 15), options, Carrier.DHL
        )
        assertEquals(0, totalCost.compareTo(cost))
    }

    @Description("Carrier does not influence shipping cost in the current implementation.")
    @TableTest("""
        Scenario     | Carrier             | Total Cost?
        Any carrier  | {DHL, UPS, FEDEX}   | 12.00
        """)
    fun carrierDoesNotAffectShippingCost(carrier: Carrier, totalCost: BigDecimal) {
        val cost = calculator.calculateShippingCost(
            ShippingZone("EU", "express"), 3.0, listOf(30, 20, 15), PackageOptions(), carrier
        )
        assertEquals(0, totalCost.compareTo(cost))
    }
}
